model/src/test/kotlin/utils/FindingsMatcherTest.kt

```diff
@@ -53,43 +56,67 @@ private fun createCopyrightFinding(statement: String, path: String, line: Int =
         )
     )
 
+private fun Collection<LicenseFindings>.getFindings(license: String) = single { it.license == license }
+
 class FindingsMatcherTest : WordSpec() {
     private val matcher = FindingsMatcher(LicenseFileMatcher("a/LICENSE"))
 
     init {
-        "getRootLicense()" should {
-            "return the license of the file matched by the license file matcher" {
+        "Given a license finding in a license file and a copyright finding in a file without license, match" should {
+            "associate that copyright with the root license" {
                 val licenseFindings = listOf(
-                    createLicenseFinding(license = "abc", path = "a/LICENSE")
+                    createLicenseFinding(license = "some id", path = "a/LICENSE")
+                )
+                val copyrightFindings = listOf(
+                    createCopyrightFinding(statement = "some stmt", path = "some/other/file")
                 )
 
-                matcher.getRootLicense(licenseFindings) shouldBe "abc"
+                val result = matcher.match(licenseFindings, copyrightFindings)
+
+                result.size shouldBe 1
+                val findings = result.getFindings("some id")
+                findings.locations.map { it.path } shouldContainExactlyInAnyOrder listOf("a/LICENSE")
+                findings.copyrights.map { it.statement } shouldContainExactlyInAnyOrder listOf("some stmt")
             }
+        }
 
-            "return an empty string when no file is matched by the license file matcher" {
-                val licenseFindings = listOf(
-                    createLicenseFinding(license = "abc", path = "b/LICENSE")
+        "Given no license finding in any license file and a copyright finding in a file without license, match" should {
+            "discard that copyright" {
+                val licenseFindings = listOf<LicenseFinding>()
```

Comment:
Would it make sense to add a license finding in a non-license file to verify that the copyright is not accidentally associated to this?

Reply:
Or maybe add a separate test for this in your last commit?

Reply:
Done. Added separate test as suggested.